services/ux-backend/handlers/devicefinder/handler.go

```diff
@@ -0,0 +1,174 @@
+package devicefinder
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/red-hat-storage/ocs-operator/v4/services/devicefinder"
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DiscoveryRequest represents the request body for device discovery
+type DiscoveryRequest struct {
+	Namespace    string               `json:"namespace"`
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	Tolerations  []corev1.Toleration  `json:"tolerations,omitempty"`
+}
+
+// DiscoveryResponse represents the response for device discovery
+type DiscoveryResponse struct {
+	Message string                     `json:"message"`
+	Success bool                       `json:"success"`
+	Devices map[string]DiscoveryResult `json:"devices,omitempty"`
+}
+
+// DiscoveryResult represents the result of device discovery for a node
+type DiscoveryResult struct {
+	DiscoveredTimeStamp string             `json:"discoveredTimeStamp,omitempty"`
+	DiscoveredDevices   []DiscoveredDevice `json:"discoveredDevices"`
+}
+
+// DiscoveredDevice shows the list of discovered devices with their properties
+type DiscoveredDevice struct {
+	DeviceID string `json:"deviceID"`
+	Path     string `json:"path"`
+	Model    string `json:"model"`
+	Type     string `json:"type"`
+	Vendor   string `json:"vendor"`
+	Size     int64  `json:"size"`
+	WWN      string `json:"WWN"`
+}
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
```

Comment:
we need to handle post and put here where the deamonset is created when all nodes are selected (default) and later users selects only the set of nodes (PUT/resource update)